### PR TITLE
Use `apply_to_collection` in `metrics_to_scalars`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #7882

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/utilities/metrics.py [hunk 1]
```diff
@@ -12,29 +12,24 @@
...
 
 import torch
 
+from pytorch_lightning.utilities.apply_func import apply_to_collection
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 
-def metrics_to_scalars(metrics: dict) -> dict:
+def metrics_to_scalars(metrics: Any) -> Any:
     """ Recursively walk through a dictionary of metrics and convert single-item tensors to scalar values. """
```
Comment (ethanwharris) [inline pytorch_lightning/utilities/metrics.py:25]: Maybe this could be updated?

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: pytorch_lightning/utilities/metrics.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/metrics.py:25]:
```suggestion
    """Recursively walk through a collection and convert single-item tensors to scalar values"""
```

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by ananthsub — APPROVED** [review]
